contents/handbook/engineering/sdks/releases.md

```diff
@@ -0,0 +1,136 @@
+---
+title: SDK releases
+sidebar: Handbook
+showTitle: true
+---
+
+This guide documents our semi-automated release process for PostHog SDKs. Each SDK repository uses a GitHub App with restricted permissions to handle releases securely, requiring team approval before any release is published.
+
+> Not all SDKs have been migrated to this process yet, but any new SDKs should definitely follow this approach. The Security folks are working on bringing all SDKs up to speed but that will take a while since we have so many of them.
+
+## How it works
+
+Our SDK release process uses a dedicated GitHub App per repository that can push directly to the main branch (bypassing branch protections) while still requiring human approval through GitHub Environments. This gives us:
+
+- **Security:** The app only has access to the specific repository it needs
+- **Auditability:** All releases require approval from the Client Libraries team
+- **Automation:** Changelog generation, version bumping, and publishing are handled automatically
+
+## Setting up releases for a new SDK
+
+When creating a new SDK - or migrating an existing one to the new workflow -, follow these steps to set up the release infrastructure.
+
+### 1. Create a GitHub App
+
+Go to [GitHub App settings](https://github.com/organizations/PostHog/settings/apps) and create a new app:
+
+1. **Name:** `Releaser (<sdk_name>)` (e.g., `Releaser (posthog-go)`)
+2. **Description:** Should be "Used to release new versions of `posthog-<sdk_name>` (e.g. "Used to release new versions of `posthog-go`.")
+3. **Homepage URL:** Point to the SDK's docs page on posthog.com (e.g., `https://posthog.com/docs/libraries/go`)
+4. **Webhook:** Disable (uncheck "Active")
+5. **Permissions:** Under "Repository permissions", set only:
+   - `Contents`: Read and write
+6. Click **Create GitHub App**
+
+After creating the app:
+
+1. Upload the following image as the app icon and set the background color to `#D97148`
+
+![SDK Releaser bot icon](https://res.cloudinary.com/dmukukwp6/image/upload/q_auto,f_auto/2514924_640cf3f534.png)
+
+2. Generate a private key and save it locally — you'll need it later
```

Comment:
```suggestion
2. Generate a private key (the option is available after creating the app). It will be downloaded locally and you should then add it to PostHog 1password as something like "Releaser Private Key <sdk_name>"
```